app/src/lib.rs

```diff
@@ -636,6 +664,33 @@ impl PortalApp {
 }
 
 impl PortalApp {
+    /// Set up relay status monitoring in a separate task
+    fn setup_relay_status_monitoring(
+        runtime: Arc<BindingsRuntime>,
+        mut notifications: tokio::sync::broadcast::Receiver<MonitorNotification>,
+        relay_status_listener: Arc<dyn RelayStatusListener>,
+    ) {
+        let runtime_clone = Arc::clone(&runtime);
+        let listener = Arc::clone(&relay_status_listener);
+        let _ = runtime_clone.add_task(async move {
+            while let Ok(notification) = notifications.recv().await {
+                match notification {
+                    MonitorNotification::StatusChanged { relay_url, status } => {
+                        // log::info!("Relay {:?} status changed: {:?}", relay_url, status);
+
```

Comment:
Drafted because here if you add a new task to call the listener, it never executes.